starter_code/app.js

```diff
@@ -1,26 +1,85 @@
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
+const SpotifyWebApi = require("spotify-web-api-node");
 
 // require spotify-web-api-node package here:
 
-
-
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
-
-
-// setting the spotify-api goes here:
-
-
-
-
-
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
+hbs.registerPartials(__dirname + "/views/partials");
+
+const clientId = "292e81d7af4a4d15afed6b919438fa84",
+  clientSecret = "c3dccb4e5842429bad289e7a890d66d7";
+
+const spotifyApi = new SpotifyWebApi({
+  clientId: clientId,
+  clientSecret: clientSecret
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then(data => {
+    spotifyApi.setAccessToken(data.body["access_token"]);
+  })
+  .catch(error => {
+    console.log("Something went wrong when retrieving an access token", error);
+  });
 
 // the routes go here:
-
-
-
-app.listen(3000, () => console.log("My Spotify project running on port 3000 🎧 🥁 🎸 🔊"));
+app.get("/", (req, res) => {
+  res.render("index");
+});
+
+app.get("/artists", (req, res) => {
+  // console.log(req.query.artistName);
+  spotifyApi
+    .searchArtists(req.query.artistName)
+    .then(data => {
+      // console.log("The received data from the API: ", data.body);
+      let artists = data.body.artists.items;
```

Comment:
Here you can create a response.locals instead of a let variable. Response local variables scoped to the request, and therefore available only to the view(s) rendered during that request / response cycle. More efficient